app/main.py

```diff
@@ -1,3 +1,80 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int):
+        self.km = km
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        if isinstance(other, (int, float)):
+            return Distance(km = self.km + other)
+        elif isinstance(other, Distance):
+            return Distance(km = self.km + other.km)
+        else:
+            return self
```

Comment:
Returning `self` for unsupported types in `__add__` is not appropriate. Consider raising a `TypeError` to indicate that the operation is not supported for the given type.